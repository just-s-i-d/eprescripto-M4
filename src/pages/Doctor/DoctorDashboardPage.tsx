import { useEffect } from "react";

import NextPatientCard from "@components/Doctor/NextPatientCard";
import InfoCards from "@components/Doctor/InfoCards";
import ErrorBoundary from "@components/ErrorBoundary";
import LineChart from "@components/Charts/LineChart";
import BarChart from "@components/Charts/BarChart";
import AppoinmentsCard from "@components/Doctor/AppoinmentsCard";
import DoughnutChart from "@components/Charts/DoughnutChart";
import {
  appointmentsEndPoint,
  barChartDataEndPoint,
  doughnutChartDataEndPoint,
  getData,
  getDataForLineGraph,
  infoCardDataEndPoint,
  lineChartDataEndPoint,
} from "@utils/Doctor";
import useStatesHook from "../../hooks/useStatesHook";
import {
  ApiResponseData,
  ApiResponseDataType,
  AppointmentDataType,
  ChartDataType,
  InfoCardDetailsType,
  LineChartApiResDataType,
} from "@constants/types";

const DoctorDashboardPage: React.FC = () => {
  const infoCards = useStatesHook<ApiResponseData<InfoCardDetailsType>>();
  const appointments = useStatesHook<ApiResponseData<AppointmentDataType>>();
  const doughnutChart = useStatesHook<ApiResponseDataType<ChartDataType>>();
  const barChart = useStatesHook<ApiResponseDataType<ChartDataType>>();
  const lineChart =
    useStatesHook<ApiResponseDataType<LineChartApiResDataType>>();
  useEffect(() => {
    getData(infoCardDataEndPoint)
      .then((res) => {
        infoCards.setData(res);
        infoCards.setError(false);
      })
      .catch(() => {
        infoCards.setError(true);
      });
  }, [infoCards.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(appointmentsEndPoint)
      .then((res) => {
        appointments.setData(res);
        appointments.setLoading(false);
        appointments.setError(false);
      })
      .catch(() => {
        appointments.setError(true);
        appointments.setLoading(false);
      });
  }, [appointments.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(doughnutChartDataEndPoint)
      .then((res) => {
        doughnutChart.setData(res);
        doughnutChart.setLoading(false);
        doughnutChart.setError(false);
      })
      .catch(() => {
        doughnutChart.setError(true);
      });
  }, [doughnutChart.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(lineChartDataEndPoint)
      .then((res) => {
        lineChart.setData(res);
        lineChart.setError(false);
      })
      .catch(() => {
        lineChart.setError(true);
      });
  }, [lineChart.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(barChartDataEndPoint)
      .then((res) => {
        barChart.setData(res);
        barChart.setError(false);
      })
      .catch(() => {
        barChart.setError(true);
      });
  }, [barChart.refresh]); //eslint-disable-line

  return (
    <>
      <ErrorBoundary>
        <div className="max-w-full flex">
          <div className="w-full flex flex-col gap-6">
            <ErrorBoundary>
              <InfoCards cardDetails={infoCards.data} infoCards={infoCards} />
            </ErrorBoundary>
            <div className="flex flex-wrap justify-between gap-y-6 max-tablet:justify-center">
              <ErrorBoundary>
                <NextPatientCard
                  appointments={appointments}
                  appointment={appointments.data && appointments.data[0]}
                  error={appointments.error}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <DoughnutChart
                  doughnutChart={doughnutChart}
                  chartData={doughnutChart.data}
                  title="Ratings Composition"
                  label="Reviews"
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <AppoinmentsCard
                  appointmentsState={appointments}
                  appointments={appointments.data}
                  loading={appointments.loading}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <LineChart
                  lineChart={lineChart}
                  chartData={lineChart.data}
                  title="Patients in Last 7 Days"
                  xAxesTitle="Dates"
                  yAxesTitle="Number of Patients"
                  getDataForLineGraph={getDataForLineGraph}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <BarChart
                  barChart={barChart}
                  chartData={barChart.data}
                  title="Age Groups Comparison"
                  label="Patients"
                  xAxesTitle="Different Age Groups (In Years)"
                  yAxesTitle="Number of Patients"
                />
              </ErrorBoundary>
            </div>
          </div>
        </div>
      </ErrorBoundary>
    </>
  );
};

export default DoctorDashboardPage;
