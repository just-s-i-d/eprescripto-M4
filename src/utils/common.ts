import { notification } from "antd";

export const showToast = (message: string, type: "success" | "error") => {
  notification.open({
    message: type === "success" ? "Success" : "Error",
    description: message,
    type: type,
    className: `bg-${type}-bgColor border rounded border-${type}-borderColor`,
    duration: 3,
  });
};
