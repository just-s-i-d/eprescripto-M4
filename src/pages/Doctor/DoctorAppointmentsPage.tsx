import { useEffect, useState } from "react";

import { ColumnsType } from "antd/es/table";
import { Card, Tooltip } from "antd";
import { PhoneTwoTone, CloseCircleTwoTone } from "@ant-design/icons";

import ErrorBoundary from "@components/ErrorBoundary";
import useStatesHook from "@hooks/useStatesHook";

import {
  appointmentsEndPoint,
  cancelAppointment,
  formatDateReadable,
  getData,
} from "@utils/Doctor";
import {
  ApiResponseData,
  AppointmentDataType,
  AppointmentType,
} from "@constants/types";
import TableCard from "@components/ui/TableCard";
import PopModal from "@components/ui/PopModal";
import { showToast } from "@utils/common";

const DoctorAppointmentsPage = () => {
  const appointmentsTable = useStatesHook<AppointmentDataType[]>();
  const [open, setOpen] = useState(false);
  const [selectedPatient, setSelectedPatient] = useState<AppointmentDataType>();
  const getDataForAppointments = () => {
    getData<ApiResponseData<AppointmentType>>(appointmentsEndPoint)
      .then((res) => {
        const data = res.map(({ id, attributes }) => {
          return {
            id,
            ...attributes,
          };
        });
        appointmentsTable.setData(data);
        appointmentsTable.setError(false);
        appointmentsTable.setLoading(false);
      })
      .catch(() => {
        appointmentsTable.setError(true);
        appointmentsTable.setLoading(false);
      });
  };
  const handlerCancleAppointment = (patient: AppointmentDataType) => {
    setOpen(true);
    setSelectedPatient(patient);
  };
  const confirmHandler = async () => {
    if (selectedPatient) {
      cancelAppointment(selectedPatient)
        .then((res) => {
          showToast(res.message, res.type);
          getDataForAppointments();
        })
        .catch((error) => {
          showToast(error.message, error.type);
        });
    }
  };
  useEffect(() => {
    getDataForAppointments();
  }, []);
  const doctorAppointmentsTableColumns: ColumnsType<AppointmentDataType> = [
    {
      title: "Patient Id",
      dataIndex: "pId",
      width: 50,
    },
    {
      title: "Patient Name",
      dataIndex: "pName",
      width: 150,
    },
    {
      title: "Email",
      dataIndex: "email",
      width: 200,
    },
    {
      title: "Date",
      dataIndex: "date",
      width: 150,
      render: (date, record) => (
        <div className="flex flex-col">
          <span>{formatDateReadable(date)}</span>
          <span>{record.timeSlot}</span>
        </div>
      ),
    },
    {
      title: "Referrer",
      dataIndex: "referrer",
      render: (referrer: string) => referrer || "NA",
      width: 100,
    },
    {
      title: "Status",
      dataIndex: "status",
      width: 50,
      render: (value: string) => (
        <span className={`py-1 px-2 rounded-lg ${value.toLowerCase()}`}>
          {value}
        </span>
      ),
      filters: [
        { text: "Pending", value: "Pending" },
        { text: "Attended", value: "Attended" },
        { text: "Cancelled", value: "Cancelled" },
      ],
      onFilter: (value, record) => record.status === value,
    },
    {
      title: "Action",
      width: 100,
      render: (_, record) => (
        <div className="flex gap-4 text-lg">
          <Tooltip title={record.contact}>
            <PhoneTwoTone
              className="rotate-90 cursor-pointer fill-[#19504c]"
              href={`tel:${record.contact}`}
            />
          </Tooltip>
          {record.status === "Pending" && (
            <CloseCircleTwoTone
              data-testid="cancel-appointment-btn"
              className="cursor-pointer"
              onClick={() => handlerCancleAppointment(record)}
            />
          )}
        </div>
      ),
    },
  ];
  return (
    <>
      <ErrorBoundary>
        <Card className="appointment-table-card h-min w-full overflow-scroll">
          <TableCard
            className="min-w-[1200px]"
            setRefresh={appointmentsTable.setRefresh}
            error={appointmentsTable.error}
            columns={
              doctorAppointmentsTableColumns as ColumnsType<
                Record<string, string | number>
              >
            }
            tableData={
              appointmentsTable.data as Record<string, string | number>[]
            }
            pageSize={7}
            setLoading={appointmentsTable.setLoading}
            loading={appointmentsTable.loading}
          />

          <PopModal
            setOpen={setOpen}
            open={open}
            okButtonText="Sure"
            title="Cancel Appointment"
            footer={true}
            confirmHandler={confirmHandler}
          >
            Are you sure,you want to cancel the appoinment for{" "}
            {selectedPatient?.pName}
          </PopModal>
        </Card>
      </ErrorBoundary>
    </>
  );
};
export default DoctorAppointmentsPage;
