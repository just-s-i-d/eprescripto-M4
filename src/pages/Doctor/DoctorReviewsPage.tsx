import { useEffect } from "react";

import { Card, Table } from "antd";
import { ColumnsType } from "antd/es/table";

import { ApiResponseData, ReviewDataType } from "@constants/types";
import { getData, reviewsDataEndPoint } from "@utils/Doctor";
import useStatesHook from "../../hooks/useStatesHook";
import starPng from "@assets/star.png";
import ErrorBoundary from "@components/ErrorBoundary";

const DoctorReviewsPage = () => {
  const reviews = useStatesHook<Record<string, string | number>[]>();
  const columns: ColumnsType<ReviewDataType> = [
    {
      title: "S. No",
      render: (_, _1, index) => index + 1,
    },
    {
      title: "Rating",
      dataIndex: "rating",
      key: "rating",
      render: (rating: number) => (
        <span className="flex">
          {Array.from({ length: rating }, (_, index) => (
            <img
              src={starPng}
              key={index}
              role="img"
              aria-label="star"
              width={20}
            />
          ))}
        </span>
      ),
      filters: [
        {
          text: "1 Star",
          value: 1,
        },
        {
          text: "2 Star",
          value: 2,
        },
        {
          text: "3 Star",
          value: 3,
        },
        {
          text: "4 Star",
          value: 4,
        },
        {
          text: "5 Star",
          value: 5,
        },
      ],
      onFilter: (value, record) => record.rating === value,
      sorter: {
        compare: (a, b) => a.rating - b.rating,
        multiple: 3,
      },
    },
    {
      title: "Comment",
      dataIndex: "comment",
      key: "comment",
    },
  ];
  useEffect(() => {
    getData<ApiResponseData<Record<string, string | number>>>(
      reviewsDataEndPoint,
    )
      .then((res) => {
        const data = res.map((element) => element.attributes);
        reviews.setData(data);
        reviews.setError(false);
        reviews.setLoading(false);
      })
      .catch(() => {
        reviews.setError(true);
        reviews.setLoading(false);
      });
  }, [reviews.refresh]);
  return (
    <>
      <Card className="overflow-scroll min-h-[50vh]">
        <ErrorBoundary
          error={reviews.error}
          refreshComponent={() => reviews.setRefresh((prev) => !prev)}
        >
          <Table
            className="custom-table min-w-[800px]"
            columns={columns as ColumnsType<Record<string, string | number>>}
            dataSource={reviews.data}
            loading={reviews.loading}
          />
        </ErrorBoundary>
      </Card>
    </>
  );
};
export default DoctorReviewsPage;
