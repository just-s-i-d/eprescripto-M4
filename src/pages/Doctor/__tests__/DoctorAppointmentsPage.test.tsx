import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import DoctorAppointmentsPage from "../DoctorAppointmentsPage";
import "@testing-library/jest-dom";
import "@test/matchMedia";
import { cancelAppointment, getData } from "@utils/Doctor";
import { showToast } from "@utils/common";

jest.mock("@utils/Doctor", () => ({
  getData: jest.fn().mockResolvedValue([
    {
      id: 8,
      attributes: {
        pId: 1,
        pName: "John Doe",
        email: "john.doe@example.com",
        date: "12/20/23",
        timeSlot: "10:00 AM - 11:00 AM",
        referrer: "Dr. Smith",
        status: "Pending",
        contact: "123-456-7890",
      },
    },
    {
      id: 9,
      attributes: {
        pId: 2,
        pName: "Jane Smith",
        email: "jane.smith@example.com",
        date: "12/20/23",
        timeSlot: "11:30 AM - 12:30 PM",
        referrer: "Dr. Johnson",
        status: "Attended",
        contact: "987-654-3210",
      },
    },
  ]),
  cancelAppointment: jest
    .fn()
    .mockResolvedValue({ message: "Cancelled", type: "success" }),
  formatDateReadable: jest.fn().mockReturnValue("28 December 2023"),
}));

jest.mock("@utils/common", () => ({
  showToast: jest.fn(),
}));

describe("Doctor Appointments Page tests", () => {
  it("renders the component and displays appointments", async () => {
    const { asFragment } = render(<DoctorAppointmentsPage />);
    await waitFor(() => {
      expect(screen.getByText("Jane Smith")).toBeInTheDocument();
      expect(screen.getByText("John Doe")).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("Cancel appointment successfull test", async () => {
    render(<DoctorAppointmentsPage />);
    await waitFor(() => {
      const cancelAppointmentBtn = screen.getByTestId("cancel-appointment-btn");
      expect(cancelAppointmentBtn).toBeInTheDocument();
      fireEvent.click(cancelAppointmentBtn);
      const confirmBtn = screen.getByText("Sure");
      expect(confirmBtn).toBeInTheDocument();
      fireEvent.click(confirmBtn);
    });
    await waitFor(() => {
      expect(cancelAppointment).toHaveBeenCalled();
    });
    await waitFor(() => {
      expect(showToast).toHaveBeenCalledWith("Cancelled", "success");
    });
  });
  it("Cancel appointment failed test", async () => {
    (cancelAppointment as jest.Mock).mockRejectedValue({
      message: "Error",
      type: "error",
    });
    render(<DoctorAppointmentsPage />);
    await waitFor(() => {
      const cancelAppointmentBtn = screen.getByTestId("cancel-appointment-btn");
      expect(cancelAppointmentBtn).toBeInTheDocument();
      fireEvent.click(cancelAppointmentBtn);
      const confirmBtn = screen.getByText("Sure");
      expect(confirmBtn).toBeInTheDocument();
      fireEvent.click(confirmBtn);
    });
    await waitFor(() => {
      expect(cancelAppointment).toHaveBeenCalled();
    });
    await waitFor(() => {
      expect(showToast).toHaveBeenCalledWith("Error", "error");
    });
  });

  it("Error state handle test", async () => {
    (getData as jest.Mock).mockRejectedValue("Mocked result");
    render(<DoctorAppointmentsPage />);
    await waitFor(() => {
      expect(screen.getByText("Something went wrong"));
    });
  });
});
