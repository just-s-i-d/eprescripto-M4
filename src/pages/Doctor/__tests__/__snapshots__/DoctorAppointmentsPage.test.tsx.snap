// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor Appointments Page tests renders the component and displays appointments 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered appointment-table-card h-min w-full overflow-scroll css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="marginTop-12 minHeight-74"
      >
        <div
          class="margin-bottom-12"
        >
          <span
            class="ant-input-affix-wrapper custom-search css-dev-only-do-not-override-1adbn6x"
          >
            <span
              class="ant-input-prefix"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              class="ant-input css-dev-only-do-not-override-1adbn6x"
              placeholder="Search"
              type="text"
              value=""
            />
          </span>
        </div>
        <div
          class="ant-table-wrapper custom-table min-w-[1200px] css-dev-only-do-not-override-1adbn6x"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
                data-testid="custom-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          style="width: 50px;"
                        />
                        <col
                          style="width: 150px;"
                        />
                        <col
                          style="width: 200px;"
                        />
                        <col
                          style="width: 150px;"
                        />
                        <col
                          style="width: 100px;"
                        />
                        <col
                          style="width: 50px;"
                        />
                        <col
                          style="width: 100px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Patient Id
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Patient Name
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Email
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Date
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Referrer
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Status
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="filter"
                                  class="anticon anticon-filter"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="filter"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Action
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            1
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            John Doe
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            john.doe@example.com
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="flex flex-col"
                            >
                              <span>
                                28 December 2023
                              </span>
                              <span>
                                10:00 AM - 11:00 AM
                              </span>
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Dr. Smith
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <span
                              class="py-1 px-2 rounded-lg pending"
                            >
                              Pending
                            </span>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="flex gap-4 text-lg"
                            >
                              <span
                                aria-label="phone"
                                class="anticon anticon-phone rotate-90 cursor-pointer fill-[#19504c]"
                                href="tel:123-456-7890"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="phone"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M721.7 184.9L610.9 295.8l120.8 120.7-8 21.6A481.29 481.29 0 01438 723.9l-21.6 8-.9-.9-119.8-120-110.8 110.9 104.5 104.5c10.8 10.7 26 15.7 40.8 13.2 117.9-19.5 235.4-82.9 330.9-178.4s158.9-213.1 178.4-331c2.5-14.8-2.5-30-13.3-40.8L721.7 184.9z"
                                    fill="#e6f4ff"
                                  />
                                  <path
                                    d="M877.1 238.7L770.6 132.3c-13-13-30.4-20.3-48.8-20.3s-35.8 7.2-48.8 20.3L558.3 246.8c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l89.6 89.7a405.46 405.46 0 01-86.4 127.3c-36.7 36.9-79.6 66-127.2 86.6l-89.6-89.7c-13-13-30.4-20.3-48.8-20.3a68.2 68.2 0 00-48.8 20.3L132.3 673c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l106.4 106.4c22.2 22.2 52.8 34.9 84.2 34.9 6.5 0 12.8-.5 19.2-1.6 132.4-21.8 263.8-92.3 369.9-198.3C818 606 888.4 474.6 910.4 342.1c6.3-37.6-6.3-76.3-33.3-103.4zm-37.6 91.5c-19.5 117.9-82.9 235.5-178.4 331s-213 158.9-330.9 178.4c-14.8 2.5-30-2.5-40.8-13.2L184.9 721.9 295.7 611l119.8 120 .9.9 21.6-8a481.29 481.29 0 00285.7-285.8l8-21.6-120.8-120.7 110.8-110.9 104.5 104.5c10.8 10.8 15.8 26 13.3 40.8z"
                                    fill="#1677ff"
                                  />
                                </svg>
                              </span>
                              <span
                                aria-label="close-circle"
                                class="anticon anticon-close-circle cursor-pointer"
                                data-testid="cancel-appointment-btn"
                                role="img"
                                tabindex="-1"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="close-circle"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                    fill="#1677ff"
                                  />
                                  <path
                                    d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm171.8 527.1c1.2 1.5 1.9 3.3 1.9 5.2 0 4.5-3.6 8-8 8l-66-.3-99.3-118.4-99.3 118.5-66.1.3c-4.4 0-8-3.6-8-8 0-1.9.7-3.7 1.9-5.2L471 512.3l-130.1-155a8.32 8.32 0 01-1.9-5.2c0-4.5 3.6-8 8-8l66.1.3 99.3 118.4 99.4-118.5 66-.3c4.4 0 8 3.6 8 8 0 1.9-.6 3.8-1.8 5.2l-130.1 155 129.9 154.9z"
                                    fill="#e6f4ff"
                                  />
                                  <path
                                    d="M685.8 352c0-4.4-3.6-8-8-8l-66 .3-99.4 118.5-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155-130.1 154.9a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3 99.3-118.5L611.7 680l66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.9 512.2l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                                    fill="#1677ff"
                                  />
                                </svg>
                              </span>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            2
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Jane Smith
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            jane.smith@example.com
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="flex flex-col"
                            >
                              <span>
                                28 December 2023
                              </span>
                              <span>
                                11:30 AM - 12:30 PM
                              </span>
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Dr. Johnson
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <span
                              class="py-1 px-2 rounded-lg attended"
                            >
                              Attended
                            </span>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="flex gap-4 text-lg"
                            >
                              <span
                                aria-label="phone"
                                class="anticon anticon-phone rotate-90 cursor-pointer fill-[#19504c]"
                                href="tel:987-654-3210"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="phone"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M721.7 184.9L610.9 295.8l120.8 120.7-8 21.6A481.29 481.29 0 01438 723.9l-21.6 8-.9-.9-119.8-120-110.8 110.9 104.5 104.5c10.8 10.7 26 15.7 40.8 13.2 117.9-19.5 235.4-82.9 330.9-178.4s158.9-213.1 178.4-331c2.5-14.8-2.5-30-13.3-40.8L721.7 184.9z"
                                    fill="#e6f4ff"
                                  />
                                  <path
                                    d="M877.1 238.7L770.6 132.3c-13-13-30.4-20.3-48.8-20.3s-35.8 7.2-48.8 20.3L558.3 246.8c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l89.6 89.7a405.46 405.46 0 01-86.4 127.3c-36.7 36.9-79.6 66-127.2 86.6l-89.6-89.7c-13-13-30.4-20.3-48.8-20.3a68.2 68.2 0 00-48.8 20.3L132.3 673c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l106.4 106.4c22.2 22.2 52.8 34.9 84.2 34.9 6.5 0 12.8-.5 19.2-1.6 132.4-21.8 263.8-92.3 369.9-198.3C818 606 888.4 474.6 910.4 342.1c6.3-37.6-6.3-76.3-33.3-103.4zm-37.6 91.5c-19.5 117.9-82.9 235.5-178.4 331s-213 158.9-330.9 178.4c-14.8 2.5-30-2.5-40.8-13.2L184.9 721.9 295.7 611l119.8 120 .9.9 21.6-8a481.29 481.29 0 00285.7-285.8l8-21.6-120.8-120.7 110.8-110.9 104.5 104.5c10.8 10.8 15.8 26 13.3 40.8z"
                                    fill="#1677ff"
                                  />
                                </svg>
                              </span>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1adbn6x"
              >
                <li
                  aria-disabled="true"
                  class="ant-pagination-prev ant-pagination-disabled"
                  title="Previous Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="left"
                      class="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a
                    rel="nofollow"
                  >
                    1
                  </a>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-next ant-pagination-disabled"
                  title="Next Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="right"
                      class="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
